src/Server.cpp

```diff
@@ -1,22 +1,98 @@
 #include "Server.hpp"
 #include "Log.hpp"
 
-Server::Server(std::string name) {
+Server::Server() : _communications(new Communications(this))
+{
 	LOG_DEBUG("Creating new server");
-	this->_name = name;
-	LOG_INFO("New server created: " << this->_name);
+	LOG_INFO("New server created");
 }
 
-Server::Server(const Server &s) {
-	this->_name = s._name;
-}
+Server::Server(const Server &s) : _clients(s._clients),
+																	_channels(s._channels),
+																	_communications(s._communications)
+{}
 
 Server &Server::operator=(const Server &s) {
-	this->_name = s._name;
+	this->_clients = s._clients;
+	this->_channels = s._channels;
+	this->_communications = s._communications;
 	return *this;
 }
 
 Server::~Server() {
 	//TODO: Check and clean clients and channels
+	delete _communications;
 	LOG_INFO("Removed server: " << this->_name);
 }
+
+void
+Server::init(int port, std::string psswd) {
+	_communications->init(port, psswd);
+}
+
+void
+Server::run() {
+	_communications->run();
+}
+
+Client&
+Server::getClient(const std::string& nickname) {
+	for (clients_iterator it = _clients.begin(); it != _clients.end(); it++)
+		if ((*it).second->getNickname() == nickname)
+			return *(*it).second;
+	return *(_clients.end()->second);
+}
+
+Client&
+Server::getClient(int fd) {
+	return *(*_clients.find(fd)).second;
+}
+
+Channel&
+Server::getChannel(const std::string& name) {
+	Channel * channel = _channels[name];
+	channel->setName(name);
+	return *channel;
+}
+
+std::vector<Client*>
+Server::getClients() {
```

Comment:
Why does this function generate a new vector instead of return the map used by the server?
Also client should be refenrences and not pointers